dashboard/app/helpers/aichat_safety_helper.rb

```diff
@@ -24,16 +25,43 @@ def find_toxicity(role, text, locale)
         comprehend_response = AichatComprehendHelper.get_toxicity(text, locale)
         return {text: text, blocked_by: 'comprehend', details: comprehend_response} if comprehend_response && comprehend_response[:toxicity] > threshold
       end
+
+      if openai_enabled?(role)
+        details = openai_safety_check(text)
+        return {text: text, blocked_by: 'openai', details: details} if details
+      end
+    end
+
+    private def openai_safety_check(text)
+      details = nil
+      # Try twice in case of network errors or model not correctly following directions and
+      # replying with something other valid expected output.
+      Retryable.retryable(tries: 2) do
+        openai_response = OpenaiChatHelper.request_safety_check(text, get_safety_system_prompt)
+        evaluation = JSON.parse(openai_response)['choices'][0]['message']['content']
+        raise "Unexpected response from OpenAI: #{evaluation}" unless VALID_EVALUATION_RESPONSES_SIMPLE.include?(evaluation)
+        if evaluation == 'INAPPROPRIATE'
+          details = {
+            text: text,
```

Comment:
Do we need to return the full text here again? We already include it as part of the full safety response